src/nvim/eval.c

```diff
@@ -22621,7 +22622,7 @@ static void term_resize(uint16_t width, uint16_t height, void *d)
   pty_process_resize(&data->proc.pty, width, height);
 }
 
-static inline void term_delayed_free(void **argv)
+static void term_delayed_free(void **argv)
 {
   TerminalJobData *j = argv[0];
   if (j->in.pending_reqs || j->out.pending_reqs || j->err.pending_reqs) {
```

Comment:
`term_delayed_free()` can now only be called from the terminal job exit callback, which means all streams are closed and there should be no more pending requests. So this `if` block could be removed.